contracts/lib/time_series/scaler.cairo

```diff
@@ -29,17 +30,46 @@ func scale_data{range_check_ptr}(
     tick_arr_len : felt,
     tick_arr : TickElem**,
     num_intervals : felt,
-) -> (output : TickElem*):
+) -> (output : TickElem**):
     alloc_locals
-    let (interval, _) = unsigned_div_rem(end_tick - start_tick, num_intervals)
-    let (output : TickElem*) = alloc()
+    let (interval, _) = unsigned_div_rem(end_tick - start_tick, num_intervals - 1)
+    let (output : TickElem**) = alloc()
     scale_data_iter(
         start_tick, end_tick, tick_arr_len, tick_arr, 0, num_intervals, interval, output, 0
     )
 
     return (output)
 end
 
+func get_bounded_tick_idx{range_check_ptr}(
+    cur_position : felt, cur_index : felt, tick_arr_len : felt, tick_arr : TickElem**
+) -> (below_index : felt, is_start : felt, is_end : felt):
+    alloc_locals
+    if cur_index == tick_arr_len:
+        return (tick_arr_len - 1, FALSE, TRUE)
+    end
+
+    # TODO: remove
```

Comment:
Leave in for now or did you forget to handle?

Reply:
yeah i think i can remove it, was left kind of superstitiously